Source/PeleLMeX_Diffusion.cpp

```diff
@@ -1100,34 +1124,32 @@ PeleLM::differentialDiffusionUpdate(
 
     // Get t^{n} data pointer
     auto* ldata_p = getLevelDataPtr(lev, AmrOldTime);
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
-#endif
-    for (amrex::MFIter mfi(advData->Forcing[lev], amrex::TilingIfNotGPU());
-         mfi.isValid(); ++mfi) {
-      const amrex::Box& bx = mfi.tilebox();
-      amrex::FArrayBox DummyFab(bx, 1);
-      auto const& rhoY_o = ldata_p->state.const_array(mfi, FIRSTSPEC);
-      auto const& fY = advData->Forcing[lev].array(mfi, 0);
-      auto const& aux_o = (m_nAux > 0)
-                            ? ldata_p->auxiliaries.const_array(mfi, 0)
-                            : DummyFab.const_array();
-      auto const& fAux = (m_nAux > 0) ? advData->Forcing_aux[lev].array(mfi, 0)
-                                      : DummyFab.array();
-      const auto dt = m_dt;
-      const auto nAux = m_nAux;
-      amrex::ParallelFor(
-        bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-          for (int n = 0; n < NUM_SPECIES; n++) {
-            fY(i, j, k, n) *= dt;
-            fY(i, j, k, n) += rhoY_o(i, j, k, n);
-          }
-          for (int n = 0; n < nAux; n++) {
-            fAux(i, j, k, n) *= dt;
-            fAux(i, j, k, n) += aux_o(i, j, k, n);
-          }
-        });
-    }
+
+    auto const& state_ma = ldata_p->state.const_arrays();
+    auto const& fY_ma = advData->Forcing[lev].arrays();
+
+    auto const& aux_ma =
+      (m_nAux > 0) ? ldata_p->auxiliaries.const_arrays() : state_ma;
+    auto const& fAux_ma =
+      (m_nAux > 0) ? advData->Forcing_aux[lev].arrays() : fY_ma;
+
+    amrex::ParallelFor(
```

Comment:
I would break this parallelfor in 2 and then increase the parallelism on numspec and numaux